parsers/ENTSOE.py

```diff
@@ -857,27 +806,22 @@ def fetch_production(zone_key, session=None, target_datetime=None,
 
     data = []
     for i in range(len(production_dates)):
-        production_values = {ENTSOE_PARAMETER_DESC[k]: v for k, v in
-                             productions[i].items()}
+        production_values = {k: v for k, v in productions[i].items()}
         production_date = production_dates[i]
 
+        production_types = {'production': {}, 'storage': {}}
+        for key in ['production', 'storage']:
+            parameter_groups = ENTSOE_PARAMETER_GROUPS[key]
+            for fuel, groups in parameter_groups.items():
+                value = sum([production_values.get(grp, 0) for grp in groups])
```

Comment:
how will that behave if all `production_values.get(grp, 0)` are None?
In the previous behavior, if all group values are None then `value` should be None (indicates that all data is missing)

Reply:
Currently it will be 0. 

Can change to be None if all grp are 0. But gas is the only one I see that behaved like what you describe, the other fuels would get 0, is that desired?

Reply:
Changed it to match what I think is the current behaviour, which is oil gets set to `None` if it is -1. Perhaps you want to improve this to something more along the lines of what you mentioned, let me know

Reply:
The oil edge case should be removed (I don't know why it was there!)
However, what we want is to make sure that if all production_values are None, then value should be None.
I think the best is probably to add an `if` that checks if all values are None and sets value to None?

Reply:
Thanks! That makes sense, I've added logic for that. Note its still possible for some very low values to get through if say one of the four coal codes has a value but I guess that's fine because it should be dealt with by `validate_production`?